File: Power Query/300-399/361/PQ_361.py
import pandas as pd

path = "Power Query/300-399/361/PQ_Challenge_361.xlsx"
input_data = pd.read_excel(path, sheet_name="Sheet1", usecols="A", nrows=21)
test = pd.read_excel(path, sheet_name="Sheet1", usecols="C:F", nrows=5)

result = (
    input_data['Log Data']
    .str.replace('Status:', '', regex=False)
    .str.extract(
        r'#(?P<Ticket_ID>\d+-\d+)#\s*'           
        r'\((?P<Priority>[A-Z]+)\)\s*'          
        r'Service:(?P<Service>[^>]+?)'          
        r'\s*>>\s*'                             
        r'(?P<Status>\w+)'                      
    )
    .assign(Service=lambda x: x['Service'].str.strip())
    .query("Status != 'Closed' and Status != 'Resolved'")
    .reset_index(drop=True)
)

print(result.equals(test))  # True

File: Power Query/300-399/361/PQ_361.R
library(tidyverse)
library(readxl)

path <- "Power Query/300-399/361/PQ_Challenge_361.xlsx"
input <- read_excel(path, sheet = "Sheet1", range = "A1:A21")
test <- read_excel(path, sheet = "Sheet1", range = "C1:F6")

pattern <- "#(?<TicketID>\\d+-\\d+)#\\s*\\((?<Priority>[A-Z]+)\\)\\s*Service:(?<Service>[^>]+)\\s>>\\s*(?<Status>\\w+)"

result = input %>%
  mutate(`Log Data` = str_remove_all(`Log Data`, "Status:"))

result = str_match(result$`Log Data`, pattern) %>%
  as_tibble() %>%
  select(-V1) %>%
  filter(!Status %in% c("Resolved", "Closed"))

all.equal(result, test, check.attributes = FALSE)
# TRUE
